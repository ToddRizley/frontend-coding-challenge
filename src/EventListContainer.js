import React, { Component } from 'react';
import Event from './Event';

class EventListContainer extends Component {

  displayEventList(events){
    if (Object.keys(events).length !== 0){
      return (events.map( (event)=> {
        return(<Event key={event.id} eventInfo={event} />)
      })
    )};
  };

  render() {
    return (
      <div className='event-list col-xs-12'>
        <div className='event-list col-xs-12'>
          <h3 className='col-xs-6'>Event List</h3>
          {this.displayEventList(this.props.currentListView)}
        </div>
      </div>
    );
  };
};

export default EventListContainer;
