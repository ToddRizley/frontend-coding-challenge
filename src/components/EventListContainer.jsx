import React, { Component } from 'react';
import Event from './Event';

export default class EventListContainer extends Component {

  displayEventList(events) {
    if (Object.keys(events).length !== 0) {
      return (events.map(event => (<Event key={event.id} eventInfo={event} />))
      );
    }
  }

  render() {
    return (
      <div className="event-list col-xs-12">
        <div className="event-list col-xs-12">
          <h3 className="col-xs-6">Event List</h3>
          {this.displayEventList(this.props.currentListView)}
        </div>
      </div>
    );
  }
}

EventListContainer.propTypes = {
  currentListView: React.PropTypes.array,
};
