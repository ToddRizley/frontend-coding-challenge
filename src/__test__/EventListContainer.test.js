import React from 'react';
import EventListContainer from '../components/EventListContainer'
import renderer from 'react-test-renderer';

describe('<EventListContainer/>', () => {
  var currentListView= [{ id: 1, title: 'Event One', start_time: '7:00 PM 11-24-16', end_time: '9:00 PM 11-24-16' }]
   it('renders correctly', () => {
    const rendered = renderer.create(
      <EventListContainer currentListView={currentListView} />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
