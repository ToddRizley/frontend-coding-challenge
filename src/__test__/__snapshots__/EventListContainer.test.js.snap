// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventListContainer/> renders correctly 1`] = `
<div
  className="event-list col-xs-12"
>
  <div
    className="event-list col-xs-12"
  >
    <h3
      className="col-xs-6"
    >
      Event List
    </h3>
    <div
      className="row col-xs-12"
    >
      Event One
       | From: 
      7:00 PM 11-24-16
       To: 
      9:00 PM 11-24-16
    </div>
  </div>
</div>
`;
